Sources/StreamChat/Workers/ThreadListUpdater.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+
+/// Makes a thread list query call to the backend and updates the local storage with the results.
+class ThreadListUpdater: Worker {
+    /// The cursor to be used for fetching a new page of threads.
+    var nextCursor: String?
```

Comment:
Ideally updater would be without any state but then the completion needs to change the type to something which contains the array of threads and the cursor. But on the other-hand, this updater is not meant to be shared anyway so it is all right.

Reply:
I agree, it should be stateless. And actually I don't see why it shouldn't be reused, especially if we make it a repository.

Reply:
The reason I put here the cursor it was because of those reasons you stated, It would require exposing the Cursor, and besides that, we use the Updaters in the State Layer, so it would be one less thing to be duplicated. But I also agree, this should not be part of the Updater